libs/datamodel/core/src/configuration/source/datasource_provider.rs

```diff
@@ -0,0 +1,13 @@
+use crate::StringFromEnvVar;
+use datamodel_connector::Connector;
+
+pub trait DatasourceProvider {
+    /// Passes the provider arg from the datasource. Must return true for all provider names it can handle.
+    fn is_provider(&self, provider: &str) -> bool;
+
+    fn canonical_name(&self) -> &str;
+
+    fn can_handle_url(&self, name: &str, url: &StringFromEnvVar) -> Result<(), String>;
+
+    fn connector(&self) -> Box<dyn Connector>;
```

Comment:
Can we add documentation here?

Reply:
What do you want to document in this case? It's returning the connector for this `DataSourceProvider`. And this is already stated in the signature.